compiler/passes/docs.cpp

```diff
@@ -295,6 +295,34 @@ std::string generateSphinxProject(std::string dirpath) {
   return std::string(moddir);
 }
 
+/* Checks for the valid Project Version (non-recursively). If an error occurs, exit and report
+ * error.
+ */
+char * checkProjectVersion(char * projectVersion) {
+  bool check = true;
+  const char * tags[] = {"alpha", "beta", "rc"};
+  for(int i=0; i<5; i++) {
+    if(i%2 != 0 && projectVersion[i] != '.') {
+      check = false;
+      break;
+    } else if(i%2 == 0 && ((int)projectVersion[i] > (int)'9' || (int)projectVersion[i] < (int)'0')) {
+      check = false;
+      break;
+    }
+  }
+  if(check && projectVersion[5] == '-') {
+    if(!(strcmp(projectVersion+6,tags[0]) == 0 || strcmp(projectVersion+6,tags[1]) == 0 || strcmp(projectVersion+6,tags[2]) == 0)) {
+      check = false;
+    }
+  }
+  if(check) {
+    return projectVersion;
+  } else {
+    USR_FATAL(astr("Invalid version format: ", projectVersion));
```

Comment:
It will be more helpful to users if you tell them why the format wasn't valid.  There's obviously a limit to what will be useful to users but not an absurd amount of effort to determine by the compiler.  In this case, maybe something like telling what character resulted in the failure?

Reply:
Added detailed error statements also